### PR TITLE
[3/3] Eng 2395 Add links to notification contents

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR adds links to notification contents. This completes all improvements for notification contents.
To achieve this, we need to port the detected server IP to aqueduct go server system. As a results, we add this additional fields to various components so that the IP is visible in both server and executor. We also add a few new interface APIs to make the IP easier to access from callers.

## Related issue number (if any)
ENG-2395

## Loom demo (together with #970 and #964 ) (if any)
Slack:
Screenshot 2023-02-08 at 4 54 00 PM

email:
Screenshot 2023-02-08 at 4 54 09 PM
Screenshot 2023-02-08 at 4 54 15 PM
Screenshot 2023-02-08 at 4 54 21 PM

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

**Comment by vsreekanti** [comment]
Contents mostly look good. Two things:
1. Before the link, the text should say "See the Aqueduct UI for more details"
2. Does it make sense to add some kind of warning for the link to make sure they don't get confused if they're not running on localhost?

**Comment by agiron123** [comment]
Maybe we should put the workflow link at the bottom?
It seems a little overwhelming to me to see a long URL at the beginning of the message.

File: src/golang/cmd/server/server/aqueduct_server.go [hunk 1]
```diff
@@ -330,3 +335,16 @@ func (s *AqServer) Restart() {
 	s.RequestMutex.Unlock()
 	s.UnderMaintenance.Store(false)
 }
+
+// Generates the full server address for display purpose.
+// If `ExternalIP` is configured, it returns `[ExternalIP]:[Port]`.
+// Otherwise it returns `<IP_ADDRESS>:[PORT]` expecting the user to replace
+// `<IP_ADDRESS>` string while using this information.
```
Comment (cw75) [inline src/golang/cmd/server/server/aqueduct_server.go:342]: "while using this information" what does this information refer to? Maybe just say "expecting the user to manually replace `<IP_ADDRESS>` with the server's IP"?

**Review by cw75 — COMMENTED** [review]
LGTM. Just a nit about a comment that I don't fully understand.

**Pull request merged**